crates/scroll/evm/src/execute.rs

```diff
@@ -0,0 +1,544 @@
+    }
+}
+
+impl<DB, EvmConfig> ScrollExecutionStrategy<DB, EvmConfig>
+where
+    EvmConfig: ConfigureEvmEnv<Header = Header>,
+{
+    /// Configures a new evm configuration and block environment for the given block.
+    ///
+    /// # Caution
+    ///
+    /// This does not initialize the tx environment.
+    fn evm_env_for_block(&self, header: &Header, total_difficulty: U256) -> EnvWithHandlerCfg {
+        let mut cfg = CfgEnvWithHandlerCfg::new(Default::default(), Default::default());
+        let mut block_env = BlockEnv::default();
+        self.evm_config.fill_cfg_and_block_env(&mut cfg, &mut block_env, header, total_difficulty);
+
+        EnvWithHandlerCfg::new_with_cfg_env(cfg, block_env, Default::default())
+    }
+}
+
+impl<DB, EvmConfig> BlockExecutionStrategy<DB> for ScrollExecutionStrategy<DB, EvmConfig>
+where
+    DB: Database<Error: Into<ProviderError> + Display>,
+    State<DB>: FinalizeExecution<Output = BundleState>,
+    EvmConfig: ConfigureEvm<Header = Header>,
+{
+    type Error = BlockExecutionError;
+
+    fn apply_pre_execution_changes(
+        &mut self,
+        block: &BlockWithSenders,
+        _total_difficulty: U256,
+    ) -> Result<(), Self::Error> {
+        // TODO (scroll): update to the Scroll chain spec
+        // TODO (scroll): update to the Curie hardfork
+        if self.chain_spec.fork(EthereumHardfork::Dao).transitions_at_block(block.number) {
+            if let Err(err) = apply_curie_hard_fork(&mut self.state) {
+                tracing::debug!(%err, "failed to apply curie hardfork");
+                return Err(ForkError::Curie.into());
+            };
+        }
+
+        Ok(())
+    }
+
+    fn execute_transactions(
+        &mut self,
+        block: &BlockWithSenders,
+        total_difficulty: U256,
+    ) -> Result<ExecuteOutput, Self::Error> {
+        let env = self.evm_env_for_block(&block.header, total_difficulty);
+        let mut evm = self.evm_config.evm_with_env(&mut self.state, env);
+
+        let mut cumulative_gas_used = 0;
+        let mut receipts = Vec::with_capacity(block.body.transactions.len());
+
+        for (sender, transaction) in block.transactions_with_sender() {
+            // The sum of the transaction’s gas limit and the gas utilized in this block prior,
+            // must be no greater than the block’s gasLimit.
+            let block_available_gas = block.header.gas_limit - cumulative_gas_used;
+            if transaction.gas_limit() > block_available_gas {
+                return Err(BlockValidationError::TransactionGasLimitMoreThanAvailableBlockGas {
+                    transaction_gas_limit: transaction.gas_limit(),
+                    block_available_gas,
+                }
+                .into())
+            }
+
+            if transaction.is_eip4844() {
+                return Err(ConsensusError::InvalidTransaction(
+                    InvalidTransactionError::Eip4844Disabled,
+                )
+                .into())
+            }
```

Comment:
maybe I also need to add 1559 and 2930 before Curie?

Reply:
Is there any handling of transaction types in revm itself?

Reply:
I'm not sure at this point. Can you create an issue `Milestone 1 Open Questions` and add this as an open question? Let's keep that as a running list of things we need to figure out.

Reply:
If we are going to keep this here then maybe we should add similar for EIP-7702 transactions.